pkg/commands/set_instancegroup.go

```diff
@@ -0,0 +1,179 @@
+	keys := make([]string, 0, len(*ks))
+	for k := range *ks {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	return strings.Join(keys, ", ")
+}
+
+// PrettyPrintKeysWithNewlines prints newline-separated keys.
+func (ks *InstanceGroupKeySetters) PrettyPrintKeysWithNewlines() string {
+	keys := make([]string, 0, len(*ks))
+	for k := range *ks {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	return strings.Join(keys, "\n ")
+}
+
+// RunSetInstancegroup implements the set instancegroup command logic.
+func RunSetInstancegroup(f *util.Factory, cmd *cobra.Command, out io.Writer, options *SetOptions) error {
+	if !featureflag.SpecOverrideFlag.Enabled() {
+		return fmt.Errorf("set instancegroup is currently feature gated; set `export KOPS_FEATURE_FLAGS=SpecOverrideFlag`")
+	}
+
+	if options.ClusterName == "" {
+		return field.Required(field.NewPath("ClusterName"), "Cluster name is required")
+	}
+	if options.InstanceGroupName == "" {
+		return field.Required(field.NewPath("InstanceGroupName"), "Instance Group name is required")
+	}
+
+	clientset, err := f.Clientset()
+	if err != nil {
+		return err
+	}
+
+	cluster, err := clientset.GetCluster(options.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	// All instance groups are needed eventually for validation, so let's grab
+	// them all and update the pointer to the one we are setting config for.
+	instanceGroups, err := ReadAllInstanceGroups(clientset, cluster)
+	if err != nil {
+		return err
+	}
+	var instanceGroupToUpdate *api.InstanceGroup
+	for _, instanceGroup := range instanceGroups {
+		if instanceGroup.GetName() == options.InstanceGroupName {
+			instanceGroupToUpdate = instanceGroup
+		}
+	}
+	if instanceGroupToUpdate == nil {
+		return fmt.Errorf("unable to find instance group with name %q", options.InstanceGroupName)
+	}
+
+	err = SetInstancegroupFields(options.Fields, instanceGroupToUpdate)
+	if err != nil {
+		return errors.Wrap(err, "unable to set instance group")
+	}
+
+	err = UpdateInstanceGroup(clientset, cluster, instanceGroups, instanceGroupToUpdate)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SetInstancegroupFields sets field values in the instance group.
+func SetInstancegroupFields(fields []string, instanceGroup *api.InstanceGroup) error {
+	validKeyToSetters := ValidInstanceGroupKeysToSetters()
+
+	for _, field := range fields {
+		kv := strings.SplitN(field, "=", 2)
+		if len(kv) != 2 {
+			return fmt.Errorf("unhandled field: %q", field)
+		}
+
+		setter, ok := validKeyToSetters[kv[0]]
+		if !ok {
+			return fmt.Errorf("unhandled field: %q; valid instancegroup keys are: %s", field, validKeyToSetters.PrettyPrintKeysWithCommas())
+		}
+
+		err := setter(instanceGroup, kv[1])
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ValidInstanceGroupKeysToSetters returns the valid keys and config setting
+// logic for instance groups.
+func ValidInstanceGroupKeysToSetters() InstanceGroupKeySetters {
```

Comment:
As I mentioned in my description, I totally agree. Would it make sense to add a few more fields with the current PR description along with more tests and then do a follow-up PR to convert bot `set instancegroup` and `set cluster` to use reflection-based config setting, or would you like that to be in this PR?

Reply:
If I were to do it I would first convert `set cluster` to reflection, then add `set instancegroup`. The reflection code should be a separate commit (or commits), if not PR.

Reply:
That totally makes sense. Does the following sound like a reasonable compromise?

1. Merge this PR with current implementation, but more fields and tests.
2. Add reflection code and move `set cluster` to use this, using original tests to help verify no regressions.
3. Move `set instancegroup` to use reflection and use original tests to help verify no regressions.

If not, I can drop/change this PR to work on the reflection code again.

Reply:
I wouldn't block that approach, not that I could anyway. The primary downside would be the cost of reviewing code that is likely to be ripped out shortly. The risk of step 2 not happening for a while means review standards on it can't be lowered.